frontend/src/metabase/lib/api.js

```diff
@@ -4,14 +4,40 @@ import querystring from "querystring";
 
 import EventEmitter from "events";
 
+import { delay } from "metabase/lib/promise";
+
 type TransformFn = (o: any) => any;
 
 export type Options = {
   noEvent?: boolean,
+  retry?: boolean,
   transformResponse?: TransformFn,
   cancelled?: Promise<any>,
   raw?: { [key: string]: boolean },
 };
+
+export type RetryOptions = {
+  maxRetries: number,
+  retryDelayIntervals?: Array<number>,
+};
+
+const NO_RETRY: RetryOptions = {
+  maxRetries: 0,
+  retryDelayIntervals: [],
+};
+
+const ONE_SECOND = 1000;
+const MAX_RETRIES = 10;
+
+const DEFAULT_RETRY: RetryOptions = {
+  maxRetries: MAX_RETRIES,
```

Comment:
The default of 10 retries + exponential backoff means it will take about 17 minutes for the request to finally fail. That seems long to me.